core/src/main/java/org/mule/routing/AbstractAggregator.java

```diff
@@ -148,27 +148,20 @@ protected Factory internalEventsGroupsObjectStoreFactory()
             @Override
             public Object create()
             {
-                try
+                ObjectStore objectStore;
+                if (persistentStores)
+                {
+                    objectStore = muleContext.getRegistry().lookupObject(OBJECT_STORE_DEFAULT_PERSISTENT_NAME);
```

Comment:
The thing is that in order to support the configurable objectStore, the partitions have to be handled a bit differently. The obtained object will be wrapped in a ProvidedPartitionableObjectStoreWrapper to handle that logic (the creation of the corresponding partitions). The type returned when using the ObjectStoreManager does not support this.